pylxd/tests/models/test_project.py

```diff
@@ -0,0 +1,166 @@
+
+
+class TestProject(testing.PyLXDTestCase):
+    """Tests for pylxd.models.Project."""
+
+    def test_get(self):
+        """A project is fetched."""
+        name = "test-project"
+        an_project = models.Project.get(self.client, name)
+
+        self.assertEqual(name, an_project.name)
+
+    def test_get_not_found(self):
+        """LXDAPIException is raised on unknown project."""
+
+        def not_found(request, context):
+            context.status_code = 404
+            return json.dumps(
+                {"type": "error", "error": "Not found", "error_code": 404}
+            )
+
+        self.add_rule(
+            {
+                "text": not_found,
+                "method": "GET",
+                "url": r"^http://pylxd.test/1.0/projects/test-project$",
+            }
+        )
+
+        self.assertRaises(
+            exceptions.LXDAPIException, models.Project.get, self.client, "test-project"
+        )
+
+    def test_get_error(self):
+        """LXDAPIException is raised on get error."""
+
+        def error(request, context):
+            context.status_code = 500
+            return json.dumps(
+                {"type": "error", "error": "Not found", "error_code": 500}
+            )
+
+        self.add_rule(
+            {
+                "text": error,
+                "method": "GET",
+                "url": r"^http://pylxd.test/1.0/projects/test-project$",
+            }
+        )
+
+        self.assertRaises(
+            exceptions.LXDAPIException, models.Project.get, self.client, "test-project"
+        )
+
+    def test_exists(self):
+        name = "test-project"
+
+        self.assertTrue(models.Project.exists(self.client, name))
+
+    def test_not_exists(self):
+        def not_found(request, context):
+            context.status_code = 404
+            return json.dumps(
+                {"type": "error", "error": "Not found", "error_code": 404}
+            )
+
+        self.add_rule(
+            {
+                "text": not_found,
+                "method": "GET",
+                "url": r"^http://pylxd.test/1.0/projects/test-project$",
+            }
+        )
+
+        name = "test-project"
+
+        self.assertFalse(models.Project.exists(self.client, name))
+
+    def test_all(self):
+        """A list of all projects is returned."""
+        projects = models.Project.all(self.client)
+
+        self.assertEqual(1, len(projects))
+
+    def test_create(self):
+        """A new project is created."""
+        an_project = models.Project.create(
+            self.client, name="new-project", config={}, devices={}
+        )
+
+        self.assertIsInstance(an_project, models.Project)
+        self.assertEqual("new-project", an_project.name)
+
+    def test_update(self):
+        """A project is updated."""
+        # XXX: rockstar (03 Jun 2016) - This just executes
+        # a code path. There should be an assertion here, but
+        # it's not clear how to assert that, just yet.
```

Comment:
It's a bit weird to add a comment from the past here :) I think it can be dropped